python/pyarrow/tests/test_pandas.py

```diff
@@ -237,6 +237,17 @@ def test_multiindex_columns_with_dtypes(self):
         df = pd.DataFrame([(1, 'a'), (2, 'b'), (3, 'c')], columns=columns)
         _check_pandas_roundtrip(df, preserve_index=True)
 
+    def test_multiindex_columns_with_tz(self):
```

Comment:
Small note on the test name: this is not actually using a MultiIndex column (it's a single level, but using a DatetimeIndex for the column names). So I would rename the test and move it up a little below where there are tests for different column index types (there seems to be already a test related to this)